packages/core/src/App/Containers/Routes/routes.jsx

```diff
@@ -68,6 +68,9 @@ const Routes = ({
     if (!lang_regex.test(location.search) && lang !== 'EN') {
         window.history.replaceState({}, document.title, urlForLanguage(lang));
     }
+    if (lang_regex.test(location.search) && lang === 'EN') {
```

Comment:
I think the idea for the above condition was to hide language from url for english since it's default. Therefore, I think we do not need to add this new condition here. Or if we now need this, you should remove the condition above and replaceState in all the cases.

Reply:
The first if statement will check if there is a language aside from english and will keep the &lang= with the appropriate language in the URL. The second if statement will remove the &lang= in the URL if the default language is english. This is done, because otherwise the user can in some cases still see the &lang= in the URL (for example the signup link). 

This code has now been refactored to make it better and more readable. Perhaps, you can have a look?